full-stack-tests/core/src/frontend/hub/ExternalTextures.test.ts

```diff
@@ -0,0 +1,97 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+import { expect } from "chai";
+import { ImageSourceFormat, RenderTexture } from "@bentley/imodeljs-common";
+import { IModelApp, IModelConnection, RemoteBriefcaseConnection } from "@bentley/imodeljs-frontend";
+import { ExternalTextureLoader, ExternalTextureRequest, GL, Texture2DHandle } from "@bentley/imodeljs-frontend/lib/webgl";
+import { TestUtility } from "./TestUtility";
+import { TestUsers } from "@bentley/oidc-signin-tool/lib/frontend";
+
+describe("external texture requests", () => {

```

Comment:
Test what happens when a request throws.

Reply:
Even with monkey-patching the private _activateRequest method, the exception would have to occur in there.
The other approach I could take is providing a test-only member on the ExternalTextureLoader class.... which forces _activateRequest to throw an exception.
Do you have any better ideas?

Reply:
One thing that should throw and definitely should be tested is requesting a texture image for an element that doesn't exist or isn't a texture element.

Reply:
Ah - right - that's a good approach.  Thanks.

Reply:
Added testing.

Reply:
Made this test operate now by specifying a non-string texture name.